### PR TITLE
Fix #2826 - Corrected handling of IEnumerable constants in parameterizer

### LINKED ISSUE #2826: Disable Caching
I'm currently facing an issue (as explained in [this stackoverflow question](http://stackoverflow.com/questions/31953837/cached-results-in-linq-query)) that I suspect may be related to caching.

What is the right way to disable caching for a specific query in EF7?

### PR DESCRIPTION
**Pull request by anpete**

### CODE REVIEW

**Comment by anpete** [comment]
@maumar

File: src/EntityFramework.Core/Query/ExpressionVisitors/ParameterExtractingExpressionVisitor.cs [hunk 1]
```diff
@@ -82,36 +83,49 @@ public override Expression Visit(Expression expression)
...
-                try
-                {
-                    string parameterName;
-                    var parameterValue = ExpressionEvaluationHelpers.Evaluate(e, out parameterName);
+                var constantExpression = e as ConstantExpression;
 
-                    var compilerPrefixIndex = parameterName.LastIndexOf(">", StringComparison.Ordinal);
-                    if (compilerPrefixIndex != -1)
+                if (constantExpression == null
+                    || (constantExpression.Type != typeof(string)
```
Comment (maumar) [inline src/EntityFramework.Core/Query/ExpressionVisitors/ParameterExtractingExpressionVisitor.cs]: byte array?

Comment (anpete) [inline src/EntityFramework.Core/Query/ExpressionVisitors/ParameterExtractingExpressionVisitor.cs]: Fixed thanks.

**Comment by maumar** [comment]
:shipit:

**Pull request merged**